app/controllers/api/chatrooms_controller.rb

```diff
@@ -1,36 +1,51 @@
 class Api::ChatroomsController < ApplicationController
   before_action :authenticate_request
+  before_action :validate_user, only: [:show, :all_messages]
 
   def index
-    @chatrooms = @current_user.chatrooms
-    @chatroom = []
-    # TODO: Why do you use this loop here?
-    # Can't you just return @current_user.chatrooms?
-    # Do you have a document defining your API responses?
-    @chatrooms.each do |chatroom|
-      c = Hash.new
-      c["chatroom_id"] = chatroom.id
-      chatroom.users.each do |user|
-        if user != @current_user
-          c["username"] = user.name
-        end
-      end
-      c["lastmessage"] = chatroom.messages.last
-      if c.has_key? 'username'
-        @chatroom << c
-      end
-    end
-    render json: {
-      chatrooms: @chatroom.to_json
-    }
+    @chatroom = @current_user.chatrooms
+    render json: @chatroom, root: 'chatrooms', each_serializer: AllChatroomSerializer, adapter: :json
```

Comment:
No need to explicitly assign the @chatroom variable here, you can render @current_user.chatrooms directly